airflow/providers/google/cloud/transfers/gcs_to_gcs.py

```diff
@@ -341,6 +345,8 @@ def _copy_source_without_wildcard(self, hook, prefix):
                 raise AirflowException(msg)
 
         for source_obj in objects:
+            if self.exact_match and (source_obj != prefix or not source_obj.endswith(prefix)):
+                continue
```

Comment:
@uranusjr That's the issue. According to this issue #22675 , treating these objects as prefix copies objects from the source bucket that has no match with the specified ```source_object```. So, checking ```not source_obj.endswith(prefix)``` makes sure we only copy the exact match and not copying all objects with the same prefix as the ```source_objext```

Reply:
Yeah. I see the point. The object store semantics is funny because it resembles filesystem one, but it is in fact object name, so "prefix" can often mean wrong things, for example:

* test_file.zip - might be an object (file) stored
* test_file.zip/another_file.zip - > might be ANOTHER object in something that looks like a "test_file.zip" folder. 

This is not like that in "real filesystem", you cannot have "file" and "folder" with the same name. That's why "exact-match" is sometimes the only choice as you have no other way to skip the "nested" object..